app/src/main/java/org/fossasia/openevent/general/event/EventDetailsFragment.kt

```diff
@@ -314,32 +310,42 @@ class EventDetailsFragment : Fragment() {
     private fun loadTicketFragment() {
         // Initialise Ticket Fragment
         val ticketFragment = TicketsFragment()
-        val bundle = Bundle()
-        bundle.putLong("EVENT_ID", eventId)
-        bundle.putString(CURRENCY, currency)
-        ticketFragment.arguments = bundle
-        findNavController(rootView).navigate(R.id.ticketsFragment, bundle, getAnimSlide())
+        Bundle().apply {
+            putLong(EVENT_ID, safeArgs.eventid)
+            putString(CURRENCY, currency)
+        }.also { bundle ->
+            ticketFragment.arguments = bundle
+            findNavController(rootView).navigate(R.id.ticketsFragment, bundle, getAnimSlide())
```

Comment:
Why are you passing bundle if safe args are used? Get the data from safe args generated class, no need to pass it while navigation.

Reply:
Tickets Fragment does not use SafeArgs, so there is no generated args class for TicketsFragment. Passing a bundle is the only way to do it right now. I have opened [another issue](https://github.com/fossasia/open-event-android/issues/1205) to migrate the entire project to use safe args. We won't have to use bundles then.

Also, I think to use safe args directly with the navigate call we have to use Navigation AAC's "Action" feature. We don't use it currently in this app, so we have to rely on passing bundles only. SafeArgs does help with parsing those bundles, though. So I guess we are gaining some benefit here, at least.

Reply:
> Tickets Fragment does not use SafeArgs, so there is no generated args class for TicketsFragment.

Then do it for both in proper way. No need to pass bundles while navigation with safe args.

Reply:
> I think to use safe args directly with the navigate call we have to use Navigation AAC's "Action" feature. We don't use it currently in this app, so we have to rely on passing bundles only. SafeArgs does help with parsing those bundles, though. So I guess we are gaining some benefit here, at least.

@iamareebjamal Your views?

Reply:
> > Tickets Fragment does not use SafeArgs, so there is no generated args class for TicketsFragment.
> 
> Then do it for both in proper way. No need to pass bundles while navigation with safe args.

I'm not sure if that's possible. We can only use the generated Args class as argument bundle to the `navigate()` call if we are using Actions from Nav AAC. Currently we just have a nav graph with fragment destinations only. We don't use actions to define paths between them currently.

It's better to convert the graph to use actions first. [This page](https://developer.android.com/topic/libraries/architecture/navigation/navigation-implementing#Connect-destinations) shows how it can be done.

I currently use Nav AAC with actions and safe Args in one of my own projects, and I think navigating via actions is a lot better than just passing the ID resource of the destination fragment.

Reply:
Actions also help us visualize the possible app flows in the Nav editor in Android Studio. If you want, I can open a separate issue for it and work on it. But before we do that, i think it's better if we first define the args for all fragments in the app in nav_graph.xml file first. I'm referring to [this issue for it](https://github.com/fossasia/open-event-android/issues/1205)

Reply:
I don't think it's true that you HAVE to use actions to use safe args. I may be wrong, but just arguments to destinations should generate helper classes to create and parse bundles into safe args

Reply:
Yes, just arguments for destinations do generate helper classes, and you can use them to generate arguments safely. However, if you are not using actions you will have to convert them to bundle in the end to pass them to the `navigate()` call.

For example, to navigate from EventsFragment to EventDetailsFragment, you can generate arguments this way: 
```
EventDetailsFragmentArgs.Builder()
                    .setEVENTID(eventID)
                    .build()
                    .toBundle()
                    .also { bundle ->
                        findNavController(rootView).navigate(R.id.eventDetailsFragment, bundle, getAnimFade())
                    }
```
Here I generate args using the Helper class, and then convert them to a bundle in the end.

If we were using Actions, though, we could do it this way: (this is a code snippet from one of my apps)

```
MovieDetailsFragmentDirections.actionMovieDetailsFragmentToActorDetailsFragment()
                .apply {
                    actorIdArg = id
                    transitionNameArg = transitionName
                }
                .also { action ->
                        findNavController().navigate(action)
                }
```

I have added arguments to the nav graph for EventDetailsFragment, but not to TicketsFragment. Hence there is no helper class for TicketsFragment, and I have to resort to using regular bundles.

I have not added args to nav graph for tickets fragment or others because I thought it was out of scope for this PR. I have opened a separate issue to migrate the entire project to use safe args [here](https://github.com/fossasia/open-event-android/issues/1205). Once this migration is complete, we can eventually transition the app to use Actions.

Reply:
Follow https://codelabs.developers.google.com/codelabs/android-navigation/#7 and see how it is done for search date while navigating to search from search time fragment.

Reply:
Yes, using nav actions is a good idea but it is complicated because here we are not navigation to one fragment from one destination only.

Reply:
@liveHarshit I agree with you on this. Actions are suitable only when your screen hierarchy is well defined.

I am currently working on updating the TicketsFragment to use SafeArgs, so that we can finally settle this problem

Reply:
I've pushed some new changes. TicketsFragment now uses SafeArgs too. I have changed the EventDetailsFragment to use the Builder method of the generated Args class of TicketsFragment.

Hope that resolves this issue 😊